### PR TITLE
Added density and combination hatching examples

### PR DESCRIPTION
**Pull request by forrestm**
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/shapes_and_collections/hatch_style_reference.py [hunk 1]
```diff
@@ -9,11 +9,45 @@
 import matplotlib.pyplot as plt
 from matplotlib.patches import Rectangle
 
-fig, axs = plt.subplots(2, 5, constrained_layout=True, figsize=(10, 5))
+fig, axs = plt.subplots(2, 5, constrained_layout=True, figsize=(9, 4.5))
 
 hatches = ['/', '\\', '|', '-', '+', 'x', 'o', 'O', '.', '*']
 
 
+def hatches_plot(ax, h):
```
Comment (jklymak) [inline examples/shapes_and_collections/hatch_style_reference.py:17]: You don't need to define this three times.

File: examples/shapes_and_collections/hatch_style_reference.py [hunk 2]
```diff
@@ -9,11 +9,45 @@
...
+    ax.axis('equal')
+    ax.axis('off')
+
+for ax, h in zip(axs.flat, hatches):
+    hatches_plot(ax, h)
+
+###############################################################################
+# Hatching patterns can be repeated to increase the density.
+
+fig, axs = plt.subplots(2, 5, constrained_layout=True, figsize=(9, 4.5))
```
Comment (jklymak) [inline examples/shapes_and_collections/hatch_style_reference.py:29]: These figures are too big for what they shows.  Suggest just use the normal width and shorter height.

File: examples/shapes_and_collections/hatch_style_reference.py [hunk 3]
```diff
@@ -9,11 +9,45 @@
...
+
+for ax, h in zip(axs.flat, hatches):
+    hatches_plot(ax, h)
+
+###############################################################################
+# Hatching patterns can be combined to create additional patterns.
+
+fig, axs = plt.subplots(2, 5, constrained_layout=True, figsize=(9, 4.5))
+
+hatches = ['//', '\\\\', '||', '--', '++', 'xx', 'oo', 'OO', '..', '**']
```
Comment (jklymak) [inline examples/shapes_and_collections/hatch_style_reference.py:48]: these are the same hatches as above.

File: examples/shapes_and_collections/hatch_style_reference.py (hunk 2 above)
Comment (forrestm) [reply examples/shapes_and_collections/hatch_style_reference.py:29]: How can I view the page that is created by Sphinx? I tried [building the docs](https://matplotlib.org/devdocs/devel/documenting_mpl.html#building-the-docs), but my computer froze up. Is that the only way? Or can you build just one page?

Comment (jklymak) [reply examples/shapes_and_collections/hatch_style_reference.py:29]: The Circle CI artifacts are created below and you can click through on those.  

Its not trivial to build an individual page, and yes building the docs locally is a pain to get set up.  You can, of course, just run the example locally, but you won't see the actual html.

But unless there is a good reason, its probably better to keep the width of the figure the default 6.4in.  And these swatches do not need to be very large.

**Comment by jklymak** [comment]
This rendered here: https://41228-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/shapes_and_collections/hatch_style_reference.html#sphx-glr-gallery-shapes-and-collections-hatch-style-reference-py

Looks fine to me.

**Pull request merged**